video/graphics.h

```diff
@@ -716,23 +716,48 @@ void setLegacyModes(bool legacy) {
 void scrollRegion(Rect * region, uint8_t direction, int16_t movement) {
 	canvas->setScrollingRegion(region->X1, region->Y1, region->X2, region->Y2);
 	if (ttxtMode) {
-    	if (direction == 3) ttxt_instance.scroll();
-  	} else {
+		if (direction == 3) ttxt_instance.scroll();
+	} else {
+		canvas->setPenColor(tbg);
+		auto moveX = 0;
+		auto moveY = 0;
 		switch (direction) {
-			case 0:	// Right
-				canvas->scroll(movement, 0);
-		  		break;
-			case 1: // Left
-				canvas->scroll(-movement, 0);
+			case 0:		// Right
+				moveX = 1;
+				break;
+			case 1:		// Left
+				moveX = -1;
+				break;
+			case 2:		// Down
+				moveY = 1;
 				break;
-			case 2: // Down
-				canvas->scroll(0, movement);
+			case 3:		// Up
+				moveY = -1;
 				break;
-			case 3: // Up
-				canvas->scroll(0, -movement);
+			case 4:		// positive X
```

Comment:
Acorn's version of this command supports directions 4-7, whose behaviour varies depending on the current `cursorBehaviour` variable, as set via `VDU 23,16`
since we have _partial_ support for `cursorBehaviour` it makes some sense to implement these values